src/diagnostics/MOM_diagnostics.F90

```diff
@@ -1121,6 +1122,25 @@ subroutine calculate_energy_diagnostics(u, v, h, uh, vh, ADp, CDp, G, GV, US, CS
     call post_data(CS%id_KE_visc, KE_term, CS%diag)
   endif
 
+  if (CS%id_KE_visc_gl90 > 0) then
+    ! Calculate the KE source from GL90 vertical viscosity [H L2 T-3 ~> m3 s-3].
+    do k=1,nz
+      do j=js,je ; do I=Isq,Ieq
+        KE_u(I,j) = uh(I,j,k) * G%dxCu(I,j) * ADp%du_dt_visc_gl90(I,j,k)
+      enddo ; enddo
+      do J=Jsq,Jeq ; do i=is,ie
+        KE_v(i,J) = vh(i,J,k) * G%dyCv(i,J) * ADp%dv_dt_visc_gl90(i,J,k)
+      enddo ; enddo
+      if (.not.G%symmetric) &
+        call do_group_pass(CS%pass_KE_uv, G%domain)
+      do j=js,je ; do i=is,ie
+        KE_term(i,j,k) = 0.5 * G%IareaT(i,j) &
+            * (KE_u(I,j) + KE_u(I-1,j) + KE_v(i,J) + KE_v(i,J-1))
```

Comment:
Could you add parentheses here to avoid ambiguous order-of-operation?  Probably something like this:
```
((KE_u + KE_u(i-1)) + (KE_v(j) + KE_v(j-1)))

Reply:
Good point! For consistency, I added the same parentheses in the calculation of other KE budget terms throughout `MOM_diagnostic` (see last commit).

Reply:
Wow, I had not realized that they were all missing parentheses.  Thanks for doing that!

Reply:
The change probably makes the build-test-regression fail, though... ?

Reply:
I can roll back the commit and we can handle it as a separate PR if that makes things cleaner.

Reply:
True, although I think it is more correct.  We may need a flag to preserve the old answers.  If you just want to put parentheses on the new diagnostic, that would be fine.

It should not affect the GFDL regression suite, since it does not verify diagnostics.

Reply:
Just to clarify: I will roll back the last commit, and instead commit a change that only adds parentheses to the new diagnostic?

Reply:
Yes, parentheses on the new diagnostic, and revert the old diagnostics.  And we'll address the other diagnostics in a future PR.

Reply:
(If I had known the others were missing them, I would not have even suggested it, sorry about that.)

Reply:
No problem at all!